### DISCUSSION TITLE
how does transform_instance_annotations work after convert_to_coco_dict due to difference in bbox_mode?

### DISCUSSION THREAD

**Discussion by wahmed991** [facebookresearch/detectron2#2942] [Q&A]
I'm using a custom dataset to train aligned bounding boxes with detectron. I've written my custom dataloader and mapper functions. To run evaluation on my custom dataset, I have to convert it to the coco json format using the function 'convert_to_coco_dict'. I've realized that this function converts the annotations to BoxMode.XYWH_ABS and then saves it. 

Now, when I train using augmentations, the augmentations are applied in the custom mapper function that I wrote. I'm assuming that for evaluation, these annotations from the json file are loaded and passed through the mapper. I'm thinking that since the original BoxMode that I'm using for my annotations is BoxMode.XYXY_ABS, the 'transform_instance_annotations' function assumes the annotation that got loaded from the json file to still be in this BoxMode.XYXY_ABS format and applied the transformations accordingly which is obviously wrong. Is there a work around this?

**Comment by ppwwyyxx** [comment]
> the 'transform_instance_annotations' function assumes the annotation that got loaded from the json file to still be in this BoxMode.XYXY_ABS

`transform_instance_annotations` supports both XYXY and XYWH format and will do a conversion internally.
